tools/asset-sync/assets.ps1

```diff
@@ -0,0 +1,552 @@
+
+    while (-not $reachedRoot){
+        $pathForManipulation, $remainder = Split-Path $pathForManipulation
+
+        $foundConfig, $reachedRoot = EvaluateDirectory -TargetPath $pathForManipulation
+    }
+
+    return $pathForManipulation
+}
+
+
+<#
+.SYNOPSIS
+Traverses up from a provided target directory to find a assets JSON, parses it, and returns the location and JSON content.
+
+.DESCRIPTION
+Traverses upwards until it hits either a `.git` folder or a `assets.json` file. Throws an exception if it can't find a assets.json before it hits root.
+
+.PARAMETER TargetPath
+Optional argument specifying the directory to start traversing up from. If not provided, current working directory will be used.
+#>
+Function ResolveAssetsJson {
+    param (
+        [Parameter(Mandatory = $false)]
+        [string] $TargetPath
+    )
+    $pathForManipulation = $TargetPath
+    $foundConfig = $false
+    $reachedRoot = $false
+
+    if(-not $TargetPath){
+        $pathForManipulation = Get-Location
+    }
+    
+    $foundConfig, $reachedRoot = EvaluateDirectory -TargetPath $pathForManipulation
+
+    while (-not $foundConfig -and -not $reachedRoot){
+        $pathForManipulation, $remainder = Split-Path $pathForManipulation
+
+        $foundConfig, $reachedRoot = EvaluateDirectory -TargetPath $pathForManipulation
+    }
+
+    if ($foundConfig){
+        $discoveredPath = Join-Path $pathForManipulation "assets.json"
+    }
+    else {
+        throw "Unable to locate assets.json"
+    }
+
+    # path to assets Json
+    $config = (Get-Content -Raw -Path $discoveredPath | ConvertFrom-Json)
+    Add-Member -InputObject $config -MemberType "NoteProperty" -Name "AssetsJsonLocation" -Value "$discoveredPath"
+
+    $relPath = AscendToRepoRoot -StartPath $discoveredPath
+    if($relPath){
+        try {
+            Push-Location $relPath
+            $relPath = Resolve-Path -Relative -Path $discoveredPath
+
+            # relpaths are returned with "./<blah>"
+            # given that, we need to get rid of it. This has possibility for bugs down the line.
+            $relPath = $relPath -replace "^(\.\/)|(\.\\)"
+        }
+        finally {
+            Pop-Location
+        }
+
+        # relative path to assets Json from within path
+        Add-Member -InputObject $config -MemberType "NoteProperty" -Name "AssetsJsonRelativeLocation" -Value $relPath
+    }
+
+    return $config
+}
+
+<#
+.SYNOPSIS
+Returns the location of the "assets" store. This should return a string of the form "<path to language repo root>/.assets."
+#>
+Function ResolveAssetStoreLocation {
+    if (-not (Test-Path $ASSETS_STORE)){
+        New-Item -Type Directory -Force -Path $ASSETS_STORE | Out-Null
+    }
+    $ASSETS_STORE = Resolve-Path $ASSETS_STORE
+
+    return $ASSETS_STORE
+}
+
+<#
+Takes an input string, returns the MD5 hash for the entire thing.
+
+.PARAMETER Input
+Any string.
+#>
+Function GetMD5Hash {
```

Comment:
Does this need to be MD5? I know in general we are supposed to avoid this algorithm now, however I guess we are not really using it for security reasons so maybe it doesn't matter as long as we don't end up with clashes.

Reply:
Yeah. This is effectively just a path shortner. From a security perspective this should not affect us at all.